r/tests/testthat/test-type.R

```diff
@@ -74,7 +74,18 @@ test_that("timestamp type passes along timezone parameter", {
 })
 
 test_that("decimal types pass along precision and scale", {
+  schema <- na_decimal32(12, 10)
```

Comment:
How are you thinking about validation with these helpers? This is technically an invalid precision but nanoarrow currently passes it through. I think other na_* also don't validate inputs.